lib/commands/general.js

```diff
@@ -1,40 +1,17 @@
 import log from '../logger';
-import utf7 from 'utf7';
 import androidHelpers from '../android-helpers';
 
-const {imap} = utf7;
-
 let extensions = {},
     commands = {},
     helpers = {};
 
-function encodeString (value, unicode) {
-  for (let i = 0; i < value.length; i++) {
-    let c = value.charCodeAt(i);
-    // if we're using the unicode keyboard, and this is unicode, maybe encode
-    if (unicode && (c > 127 || c === 38)) {
-      // this is not simple ascii, or it is an ampersand (`&`)
-      if (c >= parseInt("E000", 16) && c <= parseInt("E040", 16)) {
-        // Selenium uses a Unicode PUA to cover certain special characters
-        // see https://code.google.com/p/selenium/source/browse/java/client/src/org/openqa/selenium/Keys.java
-      } else {
-        // encode the text
-        value = imap.encode(value);
-        break;
-      }
-    }
-  }
-  return value;
-}
+commands.getPageSource = async function  () {
+  return await this.uiautomator2.jwproxy.command('/source', 'GET', {});
+};
 
 // Need to override this for correct unicode support
-commands.keys = async function (value) {
-  if (value instanceof Array) {
-    value = value.join("");
-  }
-  log.debug(`Setting text: '${value}'`);
-  value = encodeString(value, this.opts.unicodeKeyboard);
```

Comment:
Does this system not support Unicode? Or does it actually natively support it so we can do away with the ime stuff?